### ISSUE TITLE
False positive for `Mutex`, `Ok` and etc

### ISSUE THREAD

**Issue by xudong963** [intellij-rust/intellij-rust#8887]
## Environment

* **IntelliJ Rust plugin version: 0.4.172.4657-221-nightly**
* **Rust toolchain version: nightly-aarch64-apple-darwin unchanged - rustc 1.63.0-nightly (c35035cef 2022-05-30)**
* **IDE name and version: idea, IntelliJ IDEA 2022.1.1 (Ultimate Edition)**
* **Operating system: Macos M1**

## Problem description
<img width="1252" alt="image" src="https://user-images.githubusercontent.com/41979257/171129169-f6d8471c-0d59-42a4-a917-73b2791d5a24.png">

## Steps to reproduce

https://github.com/datafuselabs/databend/tree/b06cbbe2ca08689fb7228279d74bf304e0a4def7

**Comment by xudong963** [comment]
Hi, is this a bug in the newest IntelliJ or IntelliJ-rust plugin? Maybe I can use the old version of the plugin and IntelliJ?

**Comment by xudong963** [comment]
Hi, seems the network has a problem
```
2022-06-02 14:50:24,705 [ 292811] SEVERE - #o.r.c.p.w.StdlibDataFetcher - Execution failed (exit code 101).
/Users/xudong/.cargo/bin/cargo +nightly-aarch64-apple-darwin metadata --verbose --format-version 1 --all-features
stdout :     Updating crates.io index
warning: spurious network error (2 tries remaining): failed to connect to github.com: Connection refused; class=Os (2)
warning: spurious network error (1 tries remaining): failed to connect to github.com: Connection refused; class=Os (2)
error: failed to get `cfg-if` as a dependency of package `unwind v0.0.0 (/Users/xudong/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/unwind)`

Caused by:
  failed to load source for dependency `cfg-if`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to connect to github.com: Connection refused; class=Os (2)

stderr : 
org.rust.openapiext.RsProcessExecutionException$ProcessAborted: Execution failed (exit code 101).
/Users/xudong/.cargo/bin/cargo +nightly-aarch64-apple-darwin metadata --verbose --format-version 1 --all-features
stdout :     Updating crates.io index
warning: spurious network error (2 tries remaining): failed to connect to github.com: Connection refused; class=Os (2)
warning: spurious network error (1 tries remaining): failed to connect to github.com: Connection refused; class=Os (2)
error: failed to get `cfg-if` as a dependency of package `unwind v0.0.0 (/Users/xudong/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/unwind)`

Caused by:
  failed to load source for dependency `cfg-if`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to connect to github.com: Connection refused; class=Os (2)

stderr : 
	at org.rust.openapiext.CommandLineExtKt.execute(CommandLineExt.kt:112)
	at org.rust.openapiext.CommandLineExtKt.execute$default(CommandLineExt.kt:55)
	at org.rust.cargo.toolchain.tools.Cargo.execute(Cargo.kt:473)
	at org.rust.cargo.toolchain.tools.Cargo.execute$default(Cargo.kt:467)
	at org.rust.cargo.toolchain.tools.Cargo.fetchMetadata(Cargo.kt:171)
	at org.rust.cargo.project.workspace.StdlibDataFetcher.collectPackageMetadata(StandardLibrary.kt:298)
	at org.rust.cargo.project.workspace.StdlibDataFetcher.fetchStdlibData(StandardLibrary.kt:212)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fetchActualStdlib(StandardLibrary.kt:115)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fetchActualStdlib$default(StandardLibrary.kt:106)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile(StandardLibrary.kt:83)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile$default(StandardLibrary.kt:62)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:436)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt:428)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt:400)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:456)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$runWithChildProgress(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$SyncContext.runWithChildProgress(CargoSyncTask.kt:189)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:400)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$refreshedProjects$1$2.invoke(CargoSyncTask.kt:127)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$refreshedProjects$1$2.invoke(CargoSyncTask.kt:112)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:456)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress$default(CargoSyncTask.kt:447)
	at org.rust.cargo.project.model.impl.CargoSyncTask.doRun(CargoSyncTask.kt:112)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:80)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$5(CoreProgressManager.java:493)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:244)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:189)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:176)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:244)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-06-02 14:50:24,706 [ 292812] SEVERE - #o.r.c.p.w.StdlibDataFetcher - IntelliJ IDEA 2022.1.1  Build #IU-221.5591.52
2022-06-02 14:50:24,706 [ 292812] SEVERE - #o.r.c.p.w.StdlibDataFetcher - JDK: 11.0.14.1; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.
2022-06-02 14:50:24,706 [ 292812] SEVERE - #o.r.c.p.w.StdlibDataFetcher - OS: Mac OS X
2022-06-02 14:50:24,706 [ 292812] SEVERE - #o.r.c.p.w.StdlibDataFetcher - Plugin to blame: Rust version: 0.4.172.4659-221-nightly
```
